triggermail_templates.py

```diff
@@ -85,11 +79,23 @@ def run(self, edit):
                     use_dev='dev.' in template_filename,
                     generation=getattr(self, 'generation', 0),
                     variant_id=getattr(self, 'variant_id', ''),
-                    subaction=getattr(self, 'subaction', ''),
-                    file_names=file_names)
+                    subaction=getattr(self, 'subaction', ''))
+
         print(params)
-        if not use_cache:
-            params["templates"] = json.dumps(self.generate_file_map())
+
+        if use_pertinent_load:
+            file_names, file_map = self.generate_pertinent_file_names_and_map(template_filename)
+            params['file_names'] = json.dumps(file_names)
+            params['templates'] = json.dumps(file_map)
+        else:
+            params['file_names'] = json.dumps(self.generate_file_list())
+            use_cache = self.settings.get('use_cache', DEFAULT_USE_CACHE_SETTING)
+            if not use_cache:
+                params["templates"] = json.dumps(self.generate_file_map())
```

Comment:
Both branches assign `file_names` and `templates` to params; It might be cleaner to do something like:

```
if use_pertinent_load:
    file_names = ...
    templates = ...
else:
    file_names = ...
    params = ...
params['file_names'] = json.dumps(file_names)
params['templates'] = json.dumps(templates)
```

This would reduce duplication on the two branches